Package.swift

```diff
@@ -78,7 +65,7 @@ var awsCCalPlatformExcludes = [
 #if os(macOS)
 awsCCalPlatformExcludes.append("source/windows")
 awsCCalPlatformExcludes.append("source/unix")
-#elseif(Windows)
+#elseif os(Windows)
 awsCCalPlatformExcludes.append("source/darwin")
 awsCCalPlatformExcludes.append("source/unix")
 #else
```

Comment:
trivial: Sorry, every time I see this block of code in a PR it just makes me itchy

Looks like the only valid values are (from [docs](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Conditional-Compilation-Block)):
> macOS, iOS, watchOS, tvOS, Linux, Windows

maybe write this as:
```
#if !os(Windows)
awsCCalPlatformExcludes.append("source/windows")
#endif

#if !os(Linux)
awsCCalPlatformExcludes.append("source/unix")
#endif

#if !(os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
awsCCalPlatformExcludes.append("source/darwin")
#endif
```

or
```
#if os(Windows)
awsCCalPlatformExcludes.append("source/darwin")
awsCCalPlatformExcludes.append("source/unix")
#elif os(Linux)
awsCCalPlatformExcludes.append("source/darwin")
awsCCalPlatformExcludes.append("source/windows")
#else // macOS, iOS, watchOS, tvOS
awsCCalPlatformExcludes.append("source/windows")
awsCCalPlatformExcludes.append("source/unix")
#endif
```

or

change from an "include everything then carefully exclude things" model, to an "explicitly include things, use a helper function to gather up the majority of files from where we expect them to be" model

or

tell me to go away, we'll revamp this when we truly revisit supporting iOS

Reply:
i might be missing some context and previous discussions on this, but i agree this code block looks off to me as well. is there a major reason why we cant specify base of things to include and then add platform specific things to it instead of going exclusion route

Reply:
SwiftPM doesn't support something like "source/*.c" and recursively including everything. We can have a hacky solution where we can create a custom function that returns a list of files and then include platform-specific content. This is a hack, and we will have to make sure it does not have any unintended consequences.

In the long term, we should eliminate OS-specific checks as well because they don't work for cross-compiling use cases like MacOS -> Linux. However, the current SwiftPM approach is to have separate dependencies for different OSs and conditionally link with them, which is not feasible in our case. Another solution I can think of is linking to C repositories as binary targets and generating platform-specific binaries, but I am not sure how much effort this involves.
https://forums.swift.org/t/platform-specific-resources/43612/3

Reply:
For now, I have updated it to the second approach.